### DISCUSSION TITLE
OneHotEncoder transformer

### DISCUSSION THREAD

**Discussion by Yassine97HF** [alan-turing-institute/sktime#3453] [General]
Hi! i want  to ask you if there is a way to add `OnehotEndoder` transformer to one of my exogenous variables in my `ForecastingPipeline`?

**Comment by aiwalter** [comment]
Hi @Yassine97HF , you can try to use any `sklearn` encoder via `TabularToSeriesAdopter()` from `sktime`

**Reply by fkiraly** [reply]
Or, you can use pipeline building syntax. Let's say you want to apply this to one variable or multiple variables, e.g., `vars = "myvar"` (one variable) or multiple, e.g., `vars = ["myvar1", "myvar2"]`. Let's assume your other exogeneous variables are called `othervars`.

You can then apply `OneHotEncoder` to only `vars` and leaving `othervars`, to `fcst`, by

```python
(OneHotEncoder()[vars] + Id()[othervars]) ** fcst
```

**Reply by fkiraly** [reply]
(this should wrap the `OneHotEncoder` in `TabularToSeriesAdapter` automatically)

**Reply by Yassine97HF** [reply]
Thank you for the informations 🙏

**Reply by Yassine97HF** [reply]
Hi @fkiraly , i tried as you said , the following code :

```python
forecaster = make_reduction(ElasticNet())
pipeline = (OneHotEncoder()[cat_cols] + Id()[num_cols]) ** forecaster
```
and i got the following error  :
![image](https://user-images.githubusercontent.com/102746985/191629765-c26d36f7-ebeb-4d28-bdca-422e9b7461d9.png)

What schould i do ?

**Reply by fkiraly** [reply]
ah, it does not automatically wrap itself in `TabularToSeriesAdaptor`.

You could try to do TabularToSeriesAdaptor(OneHotEncoder) - I will investigate and maybe try to make it work without.

**Reply by fkiraly** [reply]
so, doing this:

```python
forecaster = make_reduction(ElasticNet())
pipeline = (TabularToSeriesAdaptor(OneHotEncoder())[cat_cols] + Id()[num_cols]) ** forecaster
```
